xrootd/client/client.go

```diff
@@ -24,30 +24,27 @@ package client // import "go-hep.org/x/hep/xrootd/client"
 
 import (
 	"context"
-	"io"
-	"net"
-	"sync/atomic"
+	"strconv"
+	"sync"
 
-	"go-hep.org/x/hep/xrootd/internal/mux"
-	"go-hep.org/x/hep/xrootd/internal/xrdenc"
+	"github.com/pkg/errors"
 	"go-hep.org/x/hep/xrootd/xrdproto"
 	"go-hep.org/x/hep/xrootd/xrdproto/auth"
 	"go-hep.org/x/hep/xrootd/xrdproto/auth/krb5"
 	"go-hep.org/x/hep/xrootd/xrdproto/auth/unix"
-	"go-hep.org/x/hep/xrootd/xrdproto/sigver"
 )
 
 // A Client to xrootd server which allows to send requests and receive responses.
 // Concurrent requests are supported.
 // Zero value is invalid, Client should be instantiated using NewClient.
 type Client struct {
-	cancel           context.CancelFunc
-	conn             net.Conn
-	mux              *mux.Mux
-	protocolVersion  int32
-	signRequirements xrdproto.SignRequirements
-	seqID            int64
-	auths            map[string]auth.Auther
+	cancel   context.CancelFunc
+	auths    map[string]auth.Auther
+	username string
+
+	initialSessionID string
+	mu               sync.RWMutex
+	sessions         map[string]*session
```

Comment:
I am wondering about the impact (semantically) of having one `internal/mux.Mux` per-session instead of per-client.
operationally, it will reduce contention on the channels of the `mux.Mux`, but won't that muddy things when a request is initiated from, say, the main one (`initialSessionID`, right now), and ends up being handled by another one?

I guess some overall, overarching, description of the session dispatch mechanism is needed somewhere.
probably as a doc-comment on the `type session`.